### ISSUE TITLE
Support for diffusive traps

### ISSUE THREAD

**Issue by RemDelaporteMathurin** [RemDelaporteMathurin/FESTIM#376]
Diffusing defects could be useful in some cases.
Thanks to the new object-oriented approach adopted in #373, adding support for diffusing traps is easier.

There's an `ExtrinsicTrap` class that could be adapted or inherited from. The `create_form_density` method could be adapted to include a diffusion term.

https://github.com/RemDelaporteMathurin/FESTIM/blob/ea231dbb65c6e68ab2a13969847a7c50362f08c8/FESTIM/concentration/traps/extrinsic_trap.py#L40-L55

**PROBLEM** ⚠️

FESTIM currently solves the following set of coupled equations
![image](https://user-images.githubusercontent.com/40028739/157411062-93baadc6-a494-4fb1-988f-f0b98468dadc.png)
where n is the trap density (empty + filled).

Implementing a diffusion term in the formulation `ExtrinsicTrap` would mean:
![image](https://user-images.githubusercontent.com/40028739/157411391-e7b78f39-c359-4699-a902-515fb09b8032.png)

FESTIM would solve the coupled set of equations and the additional equation governing n.
⚠️ **This would give unphysical results:**
1) H is trapped
2) n "goes away" by diffusion
3) trapped H remains here (cause c_t doesn't diffuse here) which gives c_t > n which is unphysical ⚠️ 

The set of (coupled) equations that would need to be solved is the following:

![image](https://user-images.githubusercontent.com/40028739/157410751-040547f5-244e-4fec-94a8-ba3d4912293c.png)

The first equation would need to be integrated to the main set of equations of FESTIM and the main FESTIM equations would need a serious rewriting...

Long story short: it is not "that" simple 😄 

@jhdark @mougenj @ehodille I would really like to discuss it with your here.
